### ISSUE TITLE
Fine tuning in a nested fashion.

### ISSUE THREAD

**Issue by RuskinManku** [UKPLab/sentence-transformers#255]
Hi, I have defined another Model class. As input it takes a batch of paragraphs. I want to convert the sentences in each paragraph to it's respective embedding and further process it. Like this
```
src_tensors=[]
for paragraph in batch_of_paragraphs:
        paragraph_tensor = self.sentence_encoder.encode(paragraph,convert_to_numpy=False)
        paragraph_tensor = torch.stack(paragraph_tensor)
        src_tensors.append(paragraph_tensor)
src_tensors=torch.stack(src_tensors) 
```
Basically I am encoding each paragraph, converting the list of tensors to a stack of all sentence tensors..so suppose we have 5 sentences in all paragraphs, torch.stack(paragraph_tensor) will give a tensor of shape (5,768) and then I'm finally converting the whole batch to a tensor, so if I have 100 such paragraphs, then tensor of size (100,5,768). Now suppose I mean pool the embedding to get a (100,768) size tensor and I further feed this to multiple Linear layers. My question is, can I train like this is an end-to-end fashion and the bert model weights get fine tuned during training? I checked the code and I am not sure if self.eval() or torch.no_grad in the function "encode" will cause a problem.

**Comment by nreimers** [comment]
Hi @RuskinManku 
This sadly will not work. encode() will not pass any gradients, i.e., BERT will not be fine-tuned.

However, you can use the forward pass of SentenceTransformers, i.e. the code must look something like this

```
model = SentenceTransformers(...)

# Create the features for the model
features = {}
for text in batch_sentences:
	sentence_features = model.get_sentence_features(text, longest_seq)

	for feature_name in sentence_features:
		if feature_name not in features:
			features[feature_name] = []
		features[feature_name].append(sentence_features[feature_name])

for feature_name in features:
	features[feature_name] = torch.cat(features[feature_name]).to(model.device)

# pass the features to the model forward
out_features = model.forward(features)

```
